src/bio/bio_buffer.c

```diff
@@ -42,6 +43,15 @@ dma_alloc_chunk(unsigned int cnt)
 	if (bio_spdk_inited) {
 		chunk->bdc_ptr = spdk_dma_malloc_socket(bytes, BIO_DMA_PAGE_SZ, NULL,
 							bio_numa_node);
+		/*
+		 * If it failed to allocate contiguous hugepages on specified numa node,
+		 * let's try to allocate from any numa node to satisfy the contiguity.
+		 *
+		 * This could mitigate the fragmentation issue at the cost of cross
+		 * NUMA memory accessing for certain chunks.
+		 */
+		if (chunk->bdc_ptr == NULL)
+			chunk->bdc_ptr = spdk_dma_malloc(bytes, BIO_DMA_PAGE_SZ, NULL);
```

Comment:
Should we bump a warning or info messages for this?

Reply:
Hmm, I'm not quite sure. A metric or warning message looks a bit overkilling to me.

Reply:
maybe just a debug level message to help with investigations

Reply:
sure, added.